tests/unit/test-api.c

```diff
@@ -1002,6 +1017,28 @@ main (void)
     jerry_cleanup ();
   }
 
+  /* Test stopping an infinite loop. */
+  if (jerry_is_feature_enabled (JERRY_FEATURE_VM_EXEC_STOP))
+  {
+    jerry_init (JERRY_INIT_EMPTY);
+    const char *inf_loop_code_src_p = "while(true) {}";
+    parsed_code_val = jerry_parse ((jerry_char_t *) inf_loop_code_src_p, strlen (inf_loop_code_src_p), false);
+    TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
+
+    int countdown = 6;
+
+    jerry_set_vm_exec_stop_callback (vm_exec_stop_callback, &countdown, 16);
```

Comment:
Before you ask: why it counts stack rollbacks? Because we throw an exception which cause a stack rollback and an active catch block catches this exception and restart executing the bzte code. However, our termination check runs again just before the code is executed, and raise another exception. At some point the stack will fully rolled back and the actual function returns with the exception. Then stack roll back start for the callee function.

Hm, function entry checks might not be needed. This could further optimize this code. But I need to think over this.